src/luarocks/cmd.lua

```diff
@@ -235,6 +235,7 @@ do
             table.insert(dirs, dir.path(cfg.home_tree, ".luarocks"))
          end
          table.insert(dirs, cfg.sysconfdir)
+         table.insert(dirs, cfg.homeconfdir)
```

Comment:
Oh, this is what line 235 was supposed to be doing! There's a mistake there, with `cfg.home_tree` instead of `cfg.home` — but indeed, I think a more correct fix would be to change that to `table.insert(dirs, cfg.homeconfdir)`.